src/auth/strategy/student.strategy.ts

```diff
@@ -0,0 +1,36 @@
+import { Strategy } from 'passport-jwt';
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { Student } from '../../student/student.entity';
+
+export interface JwtPayload {
+  id: string;
+}
+
+function cookieExtractor(req: any): null | string {
+  return req && req.cookies ? req.cookies?.jwt ?? null : null;
+}
+
+@Injectable()
+export class StudentStrategy extends PassportStrategy(Strategy, 'student') {
+  constructor() {
+    super({
+      jwtFromRequest: cookieExtractor,
+      secretOrKey:
+        'jfhsjkfhkjsf878947289378978*&(*&*(&YUHJJBHGI&#Y78937893oUO*#UIOU#*U#*U*(#UOIJJhuHUH#iuhU*#&*(&#*(&#*&',
```

Comment:
J/w, powtarza się wielokrotnie. W dodatku kwestią bezpieczeństwa mieć to inne na serwerze docelowym